### PR TITLE
Add RLP tests with authz

### PR DESCRIPTION
**Pull request by jsmolar**

### CODE REVIEW

File: testsuite/openshift/objects/rate_limit.py [hunk 1]
```diff
@@ -19,20 +20,43 @@ def create_instance(cls, openshift: OpenShiftClient, name, route: Referencable,
...
+                            "path": [{"segment": {"key": "ext_auth_data"}}, {"segment": {"key": user_id}}],
+                        },
+                    }
+                }
+            ]
+        }
+
     @modify
-    def add_limit(self, max_value, seconds, conditions: list[str] = None):
+    def add_limit(self, max_value, seconds, conditions: list[str] = None, variable=None):
```
Comment (pehala) [inline testsuite/openshift/objects/rate_limit.py:48]: I would change variable for something more verbose, like `enable_auth`

Comment (jsmolar) [reply testsuite/openshift/objects/rate_limit.py:48]: Do you want to use `enable_auth` like bool value here?

Comment (mganisin) [reply testsuite/openshift/objects/rate_limit.py:48]: I see that `variable` is element of `variables` list, but then it is "called" also `user_id` or `descriptor_key`. Moreover from the code I understand that "userID" is currently only possible value, I can't imagine current code would work with different value. So without deep knowledge I would say there should be better name and maybe for future "simple" value (str, bool) doesn't have to be enough.

File: testsuite/objects/sections.py [hunk 2]
```diff
@@ -38,6 +38,10 @@ class Identities(abc.ABC):
     def oidc(self, name, endpoint, credentials, selector, **common_features):
         """Adds OIDC identity provider"""
 
+    @abc.abstractmethod
+    def kubernetes(self, name: str, *audiences):
```
Comment (averevki) [inline testsuite/objects/sections.py:42]: Is this somehow relevant to this PR?

File: testsuite/openshift/objects/auth_config/sections.py [hunk 3]
```diff
@@ -64,6 +64,15 @@ def mtls(self, name: str, selector_key: str, selector_value: str, **common_featu
         """
         self.add_item(name, {"mtls": {"selector": {"matchLabels": {selector_key: selector_value}}}}, **common_features)
 
+    @modify
+    def kubernetes(self, name: str, *audiences):
```
Comment (averevki) [inline testsuite/openshift/objects/auth_config/sections.py:68]: Is this somehow relevant to this PR?

File: testsuite/tests/kuadrant/test_rate_limit_authz.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+"""
```
Comment (averevki) [inline testsuite/tests/kuadrant/test_rate_limit_authz.py:1]: I believe this test file should be under the limitador package

File: testsuite/tests/kuadrant/test_rate_limit_authz.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+@pytest.fixture(scope="module")
+def auth2(rhsso):
+    """Creates new RHSSO user and returns its authentication object for HTTPX"""
+    user = rhsso.realm.create_user("user2", "password", email="test@test.com")
+    return HttpxOidcClientAuth.from_user(rhsso.get_token, user=user)
+
+
+def test_multiple_iterations(client, auth, auth2):
+    """Tests that simple limit is applied successfully and works for multiple iterations"""
+    assert client.get("/get", auth=auth).status_code == 200
```
Comment (averevki) [inline testsuite/tests/kuadrant/test_rate_limit_authz.py:49]: Can these requests be sent through the `utilities.fire_requests` utility instead?

File: testsuite/objects/sections.py (hunk 2 above)
Comment (jsmolar) [reply testsuite/objects/sections.py:42]: No, this should be in a different PR.

File: testsuite/tests/kuadrant/test_rate_limit_authz.py (hunk 4 above)
Comment (jsmolar) [reply testsuite/tests/kuadrant/test_rate_limit_authz.py:1]: It is correctly placed under the Kuadrant package. The reason is that it uses both Limitador (RLP) and Authorino (OIDC identity) features.

**Review by pehala — DISMISSED** [review]

**Review by averevki — APPROVED** [review]
Can suggest to squash commits :+1:

File: testsuite/tests/kuadrant/test_rate_limit_authz.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+@pytest.fixture(scope="module")
+def auth(oidc_provider):
+    """Returns RHSSO authentication object for HTTPX"""
+    return HttpxOidcClientAuth(oidc_provider.get_token, "authorization")
+
+
+@pytest.fixture(scope="module")
+def auth2(rhsso):
+    """Creates new RHSSO user and returns its authentication object for HTTPX"""
+    user = rhsso.realm.create_user("user2", "password", email="test@test.com")
```
Comment (averevki) [inline testsuite/tests/kuadrant/test_rate_limit_authz.py:35]: In my understanding second user's email should somehow derive from the first user's mail because this test requires 2 different emails for identification e.g. `f"second-{first_user_mail}"`. `test@test.com` easily can be the first email also and test would fail

**Review by averevki — DISMISSED** [review]

File: testsuite/openshift/objects/rate_limit.py [hunk 7]
```diff
@@ -39,13 +39,15 @@ def create_instance(cls, openshift: OpenShiftClient, name, route: Referencable,
         return cls(model, context=openshift.context)
 
     @modify
-    def add_limit(self, name, limits: Iterable[Limit], when: Iterable[Rule] = None):
+    def add_limit(self, name, limits: Iterable[Limit], when: Iterable[Rule] = None, counters=None):
```
[the remark this answers is not in the document]
Comment (jsmolar) [reply testsuite/openshift/objects/rate_limit.py:42]: https://docs.kuadrant.io/kuadrant-operator/doc/reference/ratelimitpolicy/ and see "counters"

[the remark this answers is not in the document]
Comment (averevki) [reply testsuite/openshift/objects/rate_limit.py:42]: more about this:
```suggestion
    def add_limit(self, name, limits: Iterable[Limit], when: Iterable[Rule] = None, counters: list[str] = None):
```

**Review by averevki — APPROVED** [review]

**Review by pehala — APPROVED** [review]
